src/main/java/org/spongepowered/api/world/gen/WorldGenerator.java

```diff
@@ -35,16 +32,17 @@
 public interface WorldGenerator {
 
     /**
-     * Generates the chunk at the given position in the world. The position
-     * specified is the chunk position (ie. the world position divided by the
-     * chunk size). The newly generated chunk should be placed into the given
-     * {@link BlockBuffer}.
-     *
-     * @param world The world
-     * @param buffer The buffer to generate the region into
-     * @param position The chunk position
+     * Gets the {@link GeneratorPopulator} of this world generator.
+     * @return The {@link GeneratorPopulator}.
+     * @see #setGeneratorPopulator(GeneratorPopulator)
+     */
+    GeneratorPopulator getGeneratorPopulator();
```

Comment:
Using a GeneratorPopulator for the base terrain generation is good, but there is a secondary need for them which is performing operations similar to that of the standard Populators but during generation. Things similar to the generation of caves would need to be applied as GeneratorPopulators. So having a 'primary' GeneratorPopulator is good but an addition structure similar to that of Populators should also be added.